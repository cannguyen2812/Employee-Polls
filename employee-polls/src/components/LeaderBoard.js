import { connect } from "react-redux";
import "../styles/LeaderBoard.css"
function LeaderBoard(props) {
    return (
        <div className="leader-board-page">
            <table>
                <thead>
                    <tr>
                        <th data-testid="userTestId" style={{ width: '500px' }}>User</th>
                        <th data-testid="answeredTestId" style={{ width: '200px' }}>Answered</th>
                        <th data-testid="createdTestId" style={{ width: '200px' }}>Created</th>
                    </tr>
                </thead>
                <tbody>
                    {props.listUser && props.listUser.map((user) => (
                        <tr key={user.id}>
                            <td style={{ display: "flex" }}>
                                <img className="image-table"
                                    src={user.avatarURL} alt=""></img>
                                <div className="user-data">
                                    <div>{user.name}</div>
                                    <div>{user.id}</div>
                                </div>
                            </td>
                            <td className="center-data-table">{Object.keys(user.answers).length}</td>
                            <td className="center-data-table">{Array.isArray(user.questions) ? user.questions.length : Object.keys(user.questions).length}</td>
                        </tr>))}
                </tbody>
            </table>
        </div>
    )
}
const mapStateToProps = (state) => {
    const listUser = state.user ? Object.values(state.user).sort((a, b) => {
        return (Object.keys(b.answers).length + b.questions.length) - (Object.keys(a.answers).length + a.questions.length)
    }) : '';
    return {
        listUser
    }
}
export default connect(mapStateToProps)(LeaderBoard);