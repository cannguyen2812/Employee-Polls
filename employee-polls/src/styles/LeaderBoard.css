.leader-board-page {
    padding-top: 50px;
    display: flex;
    justify-content: center;
}

.center-data-table {
    text-align: center;
}

table {
    border: 1px solid;
}

thead tr th {
    border-bottom: 1px solid;
    border-right: 1px solid;
}

tbody tr td {
    margin: 10px 0 10px 0;
}

thead tr th:last-child {
    border-right: none;
}

.image-table {
    width: 45px;
    height: 43px;
    object-fit: cover;
    border-radius: 50%
}

.user-data {
    margin-left: 10px;
}

