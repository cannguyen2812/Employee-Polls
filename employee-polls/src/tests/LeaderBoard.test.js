import { render, screen } from "@testing-library/react"
import LeaderBoard from "../components/LeaderBoard"
import configureStore from "redux-mock-store"
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";

const storeToMock = configureStore([]);
const store = storeToMock({
    listUser: [
        {
            id: 'sarahedo',
            password: 'password123',
            name: 'Sarah Edo',
            avatarURL: 'https://mrwallpaper.com/images/hd/chibi-luffy-smile-k40llx1k7m550s3y.jpg',
            answers: {
                "8xf0y6ziyjabvozdd253nd": 'optionOne',
                "6ni6ok3ym7mf1p33lnez": 'optionOne',
                "am8ehyc8byjqgar0jgpub9": 'optionTwo',
                "loxhs1bqm25b708cmbf3g": 'optionTwo'
            },
            questions: ['8xf0y6ziyjabvozdd253nd', 'am8ehyc8byjqgar0jgpub9']
        },
        {
            id: 'tylermcginnis',
            password: 'abc321',
            name: 'Tyler McGinnis',
            avatarURL: 'https://ih1.redbubble.net/image.5119216764.2026/st,small,845x845-pad,1000x1000,f8f8f8.jpg',
            answers: {
                "vthrdm985a262al8qx3do": 'optionOne',
                "xj352vofupe1dqz9emx13r": 'optionTwo',
            },
            questions: ['loxhs1bqm25b708cmbf3g', 'vthrdm985a262al8qx3do'],
        },
    ],
})

describe('Leader Board page', () => {
    test('render Leader Board page with correct user', () => {
        const testRender = render(
            <MemoryRouter>
                <Provider store={store}>
                    <LeaderBoard></LeaderBoard>
                </Provider>
            </MemoryRouter>
        );
        expect(testRender).toMatchSnapshot();
    })

    test("fileds are exist", () => {
        render(
            <MemoryRouter>
                <Provider store={store}>
                    <LeaderBoard></LeaderBoard>
                </Provider>
            </MemoryRouter>
        );
        expect(screen.getByTestId('userTestId')).toBeInTheDocument();
        expect(screen.getByTestId('answeredTestId')).toBeInTheDocument();
        expect(screen.getByTestId('createdTestId')).toBeInTheDocument();
    })
})
