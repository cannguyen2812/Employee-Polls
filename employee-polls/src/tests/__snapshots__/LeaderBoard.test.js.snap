// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leader Board page render Leader Board page with correct user 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="leader-board-page"
      >
        <table>
          <thead>
            <tr>
              <th
                data-testid="userTestId"
                style="width: 500px;"
              >
                User
              </th>
              <th
                data-testid="answeredTestId"
                style="width: 200px;"
              >
                Answered
              </th>
              <th
                data-testid="createdTestId"
                style="width: 200px;"
              >
                Created
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="leader-board-page"
    >
      <table>
        <thead>
          <tr>
            <th
              data-testid="userTestId"
              style="width: 500px;"
            >
              User
            </th>
            <th
              data-testid="answeredTestId"
              style="width: 200px;"
            >
              Answered
            </th>
            <th
              data-testid="createdTestId"
              style="width: 200px;"
            >
              Created
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
